__tests__/endpoint-requests.test.js

```diff
@@ -169,3 +169,66 @@ describe("GET /api/articles/:article_id/comments", () => {
       });
   });
 });
+
+describe("POST /api/articles/:article_id/comments", () => {
+  test("status: 200, should post an object to comments with the specified properties", () => {
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send({ username: "butter_bridge", body: "Hello world" })
+      .expect(201)
+      .then((response) => {
+        const comment = response.body;
+        expect(comment).toEqual(
+          expect.objectContaining({
+            comment: "Hello world",
+          })
+        );
+      })
+      .then(() => {
+        return request(app)
+          .get("/api/articles/2/comments")
+          .expect(200)
+          .then((response) => {
+            expect(response.body.comments).toHaveLength(1);
+          });
+      });
+  });
+  test("status 400: should return bad request when body is missing required fields", () => {
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send({ body: "Hello world" })
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe("400 Error - Bad Request");
+      });
+  });
+  test("status 400: should return bad request when invalid data type is passed in the request body", () => {
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send({ username: 42, body: 55 })
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe("400 Error - Bad Request");
+      });
+  });
+  test("status 400: should return bad request when user tries to post a comment to an article that does not exist ", () => {
+    return request(app)
+      .post("/api/articles/200000/comments")
```

Comment:
this test condition actually falls under the umbrella of 404 - because the data input is correct ( it hasn't broken the query ) - it just hasn't found any articles that match

Reply:
I designed my test around that and then couldn't get it ever to return 404. PSQL would always return 23502 foreign key violation which hits one of my other error handlers and returns a 400 error.

Reply:
That's interesting, it is likely due to the table setup, if article_id is a foreign key then it will throw a psql error - if 23502 is unique to that type of error then perhaps that code should be set to send back a 404